src/core/historymgr.cpp

```diff
@@ -156,16 +160,35 @@ void HistoryMgr::add(const QString &p_path,
     emit historyUpdated();
 }
 
-void HistoryMgr::insertHistoryItem(QVector<HistoryItem> &p_history, const HistoryItem &p_item)
+void HistoryMgr::remove(const QVector<QString> &p_paths, Notebook *p_notebook)
+{
+    for(const QString &p_itemPath : p_paths) {
+        if (p_notebook && !p_notebook->history()->getHistory().isEmpty()) {
```

Comment:
Judge whether we have enabled per-notebook history?

Reply:
Done.